pkg/utils/utils.go

```diff
@@ -249,6 +249,22 @@ func CustomizePodSpec(pts *corev1.PodTemplateSpec, ba common.BaseApplication) {
 		})
 	}
 
+	extraContainers := ba.GetExtraContainers()
+	if extraContainers != nil {
```

Comment:
You could make this code more efficient by using a map with `extraContainers[i].Name` as map key, and `container` definition as value. In that case, you wouldn't need the inner loop anymore ;)

Reply:
This will also not update sidecard containers after they have been appended first time.
Need to overwrite the containers on future updates  by name